kernel/kernel-api/src/test/scala/io/delta/kernel/internal/DeltaHistoryManagerSuite.scala

```diff
@@ -29,25 +29,52 @@ class DeltaHistoryManagerSuite extends AnyFunSuite with MockFileSystemClientUtil
   def checkGetActiveCommitAtTimestamp(
     fileList: Seq[FileStatus],
     timestamp: Long,
-    expectedVersion: Long): Unit = {
+    expectedVersion: Long,
+    mustBeRecreatable: Boolean = true,
+    canReturnLastCommit: Boolean = false,
+    canReturnEarliestCommit: Boolean = false): Unit = {
     val activeCommit = DeltaHistoryManager.getActiveCommitAtTimestamp(
       createMockFSListFromEngine(fileList),
       logPath,
-      timestamp
+      timestamp,
+      mustBeRecreatable,
+      canReturnLastCommit,
+      canReturnEarliestCommit
     )
-    assert(activeCommit == expectedVersion,
+    assert(activeCommit.version == expectedVersion,
       s"Expected version $expectedVersion but got $activeCommit for timestamp=$timestamp")
+
+    if (mustBeRecreatable) {
```

Comment:
As per my previous comment, I'm just curious why we need to test this extra case? What's a future use case where we want `mustBeRecreatable` to be true?

Reply:
answered above